sample_gallery/nl2sql-eval/README.md

```diff
@@ -0,0 +1,95 @@
+# Evaluate Natural Language to SQL
+
+NL2SQL (Natural Language to SQL) is a process that allows users to input queries in natural, human language and have them translated into SQL queries. This is particularly useful for users who may not be familiar with SQL syntax but need to retrieve data from a database.
+
+In this sample, we will evaluate the performance of a model that translates natural language queries into SQL queries by the metrics in the next section.
+
+## Evaluation metrics
+
+- **Exact String Match (EM)**: This metrics compares the predicted SQL query with the ground truth SQL query by comparing the strings. If the predicted SQL query is exactly the same as the ground truth SQL query, then the EM score is 1. Otherwise, the EM score is 0.
+
+- **Execution Accuracy (EX)**: This metric evaluates the execution accuracy of the predicted SQL query. The predicted SQL query is executed on the database and the results are compared with the ground truth results. The EX score is calculated as the number of correct results divided by the total number of results.
+
+- **Evaluate by LLMs**: This metric evaluates the performance of the model by comparing the predicted SQL query with the ground truth SQL query using the LLMs. The LLMs follows the prompt instruction to calculate the score.
+
+- **Vector Similiarity**: This metric evaluates the performance of the model by comparing the predicted SQL query with the ground truth SQL query using the vectors. The score is calculated by cosine similarity.
+
+## Getting Started
+
+These instructions will get you a copy of the sample up and running using PromptFlow for development and testing purposes.
+
+### Prerequisites
+
+1.  The libraries required to run this project are listed in the `requirements.txt` file. To install them, use the following command:
+
+```bash
+    pip  install  -r  requirements.txt
+```
+
+2. Deploy an Azure SQL Database with sample (AdventureWorksLT) - [Setup Guide)([Create a single database - Azure SQL Database | Microsoft Learn](https://learn.microsoft.com/en-us/azure/azure-sql/database/single-database-create-quickstart?view=azuresql&tabs=azure-portal)
+3. Install and setup the [ODBC Driver for Microsoft SQL Server (17)](https://learn.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver16#version-17) to connect to the Azure SQL Database using pyodbc
```

Comment:
Here links to ODBC version 17, but the version in 'Query.py' is ODBC Driver 18. Maybe we can modify it and refer to [the readme for nl2sql flow](https://github.com/microsoft/promptflow-resource-hub/pull/17/files#diff-4568780c7ceb2595d44673ced9617c275575e0b69d9afa44be47636e773bb22b) to add a note:
![image](https://github.com/microsoft/promptflow-resource-hub/assets/40321821/147f498b-620d-4f8b-84bb-2b900550f951)